core/src/main/java/org/bitcoinj/evolution/QuorumState.java

```diff
@@ -183,17 +186,27 @@ public GetSimplifiedMasternodeListDiff getMasternodeListDiffRequest(StoredBlock
                 nextBlock.getHeader().getHash());
     }
 
-    public ArrayList<Masternode> getAllQuorumMembers(LLMQParameters.LLMQType llmqType, Sha256Hash blockHash) {
+    public ArrayList<Masternode> getAllQuorumMembers(LLMQParameters.LLMQType llmqType, Sha256Hash quorumBaseBlockHash) {
+        StoredBlock quorumBaseBlock = getBlockFromHash(quorumBaseBlockHash);
+        return computeQuorumMembers(llmqType, quorumBaseBlock);
+    }
+
+    protected ArrayList<Masternode> computeQuorumMembers(LLMQParameters.LLMQType llmqType, StoredBlock quorumBaseBlock) {
+        boolean evoOnly = (params.getLlmqPlatform() == llmqType) && params.isV19Active(quorumBaseBlock);
         LLMQParameters llmqParameters = params.getLlmqs().get(llmqType);
-        SimplifiedMasternodeList allMns = getListForBlock(blockHash);
-        if (allMns != null) {
-            Sha256Hash modifier = LLMQUtils.buildLLMQBlockHash(llmqType, blockHash);
-            boolean hmpnOnly = (params.getLlmqPlatform() == llmqType) && params.isV19Active(allMns.getStoredBlock());
-            return allMns.calculateQuorum(llmqParameters.getSize(), modifier, hmpnOnly);
+        checkNotNull(llmqParameters);
+        if (llmqParameters.useRotation() || quorumBaseBlock.getHeight() % llmqParameters.getDkgInterval() != 0) {
+            return Lists.newArrayList();
         }
-        return null;
-    }
 
+        StoredBlock workBlock = params.isV20Active(quorumBaseBlock) ?
+                getBlockAncestor(quorumBaseBlock, quorumBaseBlock.getHeight() - 8) :
+                quorumBaseBlock;
+
+        Sha256Hash modifier = getHashModifier(llmqParameters, quorumBaseBlock);
+        SimplifiedMasternodeList allMns = getListForBlock(workBlock.getHeader().getHash());
+        return allMns != null ? allMns.calculateQuorum(llmqParameters.getSize(), modifier, evoOnly) : null;
+    }
```

Comment:
this is a refactor or non-rotated quorum membership calculation to match more closely to Core v20's code.